modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -247,109 +246,115 @@ public SCIMResponse delete(String id, UserManager userManager) {
-                    throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
-                }
-            }
-            //If a value for "sortBy" is provided and no "sortOrder" is specified, "sortOrder" SHALL default to
-            // ascending.
-            if (sortOrder == null && sortBy != null) {
-                sortOrder = SCIMConstants.OperationalConstants.ASCENDING;
-            }
 
-            // unless configured returns core-user schema or else returns extended user schema)
+            // Resolve sorting order.
+            sortOrder = resolveSortOrder(sortOrder, sortBy);
+
+            // Unless configured returns core-user schema or else returns extended user schema.
             SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
 
-            if (filter != null) {
-                filterTreeManager = new FilterTreeManager(filter, schema);
-                rootNode = filterTreeManager.buildTree();
-            }
+            // Build node for filtering.
+            Node rootNode = buildNode(filter, schema);
 
-            //obtain the json encoder
-            encoder = getEncoder();
+            // Obtain the json encoder.
+            JSONEncoder encoder = getEncoder();
 
-            //get the URIs of required attributes which must be given a value
-            Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
-                    (SCIMResourceTypeSchema)
-                            CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+            // Get the URIs of required attributes which must be given a value.
+            Map<String, Boolean> requiredAttributes = ResourceManagerUtil
+                    .getOnlyRequiredAttributesURIs((SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes,
+                            excludeAttributes);
 
-            List<Object> returnedUsers;
-            int totalResults = 0;
-            //API user should pass a usermanager usermanager to UserResourceEndpoint.
+            // API user should pass a user manager to UserResourceEndpoint.
             if (userManager != null) {
-                List<Object> tempList = userManager.listUsersWithGET(rootNode, startIndex, count,
-                        sortBy, sortOrder, domainName, requiredAttributes);
+                List<Object> tempList = userManager
+                        .listUsersWithGET(rootNode, startIndex, count, sortBy, sortOrder, domainName,
+                                requiredAttributes);
 
-                if (tempList == null) {
-                    tempList = Collections.emptyList();
-                }
+                return processUserList(tempList, encoder, schema, attributes, excludeAttributes, startIndex);
+            } else {
+                String error = "Provided user manager handler is null.";
+                // Log the error as well.
```

Comment:
We didn't log the error here? is it a redundant comment?

Reply:
fixed: https://github.com/wso2/charon/pull/263/commits/61f4e744f53695f16e5ad6c9496f369ff77101f8